metadata-ingestion/tests/integration/dbt/test_dbt.py

```diff
@@ -233,6 +233,39 @@ def test_dbt_ingest(dbt_test_config, pytestconfig, tmp_path, mock_time, requests
     )
 
 
+@pytest.mark.integration
+@freeze_time(FROZEN_TIME)
+def test_dbt_test_connection_success(pytestconfig):
+    test_resources_dir = pytestconfig.rootpath / "tests/integration/dbt"
+    config = {
+        "manifest_path": str((test_resources_dir / "dbt_manifest.json").resolve()),
+        "catalog_path": str((test_resources_dir / "dbt_catalog.json").resolve()),
+        "target_platform": "postgres",
+    }
+    report = DBTCoreSource.test_connection(config)
+    assert report is not None
+    assert report.basic_connectivity
+    assert report.basic_connectivity.capable
+    assert report.basic_connectivity.failure_reason is None
+
+
+@pytest.mark.integration
+@freeze_time(FROZEN_TIME)
+def test_dbt_test_connection_failure(pytestconfig):
+    test_resources_dir = pytestconfig.rootpath / "tests/integration/dbt"
+    config = {
+        "manifest_path": str((test_resources_dir / "dbt_manifest.json").resolve()),
+        "catalog_path": str((test_resources_dir / "dbt_cat.json").resolve()),
```

Comment:
to make this more clear
```suggestion
        "catalog_path": str((test_resources_dir / "dbt_catalog-this-file-does-not-exist.json").resolve()),
```